at-angular.ts

```diff
@@ -107,8 +144,43 @@ module at {
             if (target.$inject && target.$inject.length > 0) {
                 factory.$inject = target.$inject.slice(0);
             }
-            angular.module(moduleName).factory(className, factory);
+            angular.module(moduleName).factory(className || getFuncName(target), factory);
+        };
+    }
+
+    export interface IDecoratorAnnotation {
+        (moduleName: string, targetProvider: string): IClassAnnotationDecorator;
+    }
+
+    export function decorator(moduleName: string, targetProvider: string): at.IClassAnnotationDecorator {
```

Comment:
And it would be nice if you followed such direction in code style:

```typescript

    interface IDecoratorStatic<D> extends Function {
        new (...args: any[]): D;
    }

    export function decorator<A, B>(moduleName: string, targetProvider: string): at.IClassAnnotationDecorator {
        return (targetClass: IDecoratorStatic<B>): void => {
            angular
                .module(moduleName)
                .config(['$provide', ($provide: angular.auto.IProvideService): void => {
                    $provide.decorator(targetProvider, ['$delegate', '$injector', (
                        $delegate: A,
                        $injector: angular.auto.IInjectorService
                    ): A & B => angular.extend(
                        $delegate,
                        targetClass.prototype,
                        $injector.instantiate<B>(targetClass, {$delegate: $delegate})
                    )]);
                }]);
        };

    }
```

:D